packages/nuxt/src/app/components/client-only.ts

```diff
@@ -51,8 +53,23 @@ export function createClientOnly<T extends ComponentOptions> (component: T) {
   }
 
   clone.setup = (props, ctx) => {
+    const instance = getCurrentInstance()!
+
+    // @ts-expect-error internal
+    const inheritAttrs = instance.inheritAttrs
+    // remove exsting directives during hydration
+    const directives = extractDirectives(instance)
+    // prevent attrs inheritance since a staticVNode is rendered before hydration
+    // @ts-expect-error internal
+    instance.inheritAttrs = false
```

Comment:
not sure if it's a good idea to rely on internals

Reply:
yes, I'm not sure I would like to do that either. 🤔